webview-ui/src/context/ExtensionStateContext.tsx

```diff
@@ -142,7 +142,16 @@ export const ExtensionStateContextProvider: React.FC<{ children: React.ReactNode
 	}, [])
 
 	const handleInputChange = useCallback(
-		(field: keyof ApiConfiguration) => (event: any) => {
+		(field: keyof ApiConfiguration, softUpdate?: boolean) => (event: any) => {
+			if (softUpdate === true) {
+				setState((currentState) => {
+					return {
+						...currentState,
+						apiConfiguration: { ...currentState.apiConfiguration, [field]: event.target.value },
+					}
+				})
+				return
+			}
 			setState((currentState) => {
```

Comment:
I didn't notice this before, but why is there a setState here if we don't actually want to update the state?

Reply:
I think I'm kind of confused about this logic

Reply:
Don't know why, but the last time i use  "state" to using upsertApiConfig directly it not latest version, i check around with help from Roo, it using useState to get latest currentState to do thit